src/Image.php

```diff
@@ -381,6 +410,55 @@ public function quality($quality)
     }
 
     /**
+     * Add a watermark to current image
+     *
+     * @param string|object $watermark Image to set as watermark
+     * @param array         $settings  Overwrite default settings
+     *
+     * @return self
+     */
+    public function watermark($watermark, $settings = [])
+    {
+        if (is_string($watermark)) {
+            $watermark = self::fromFile($watermark);
+        }
+
+        $settings = array_merge(self::$watermarkSettings, $settings);
+
+        if ($settings['opacity'] < 100) {
+            $watermark->opacity($settings['opacity']);
+        }
+
+        $iw = $this->getWidth();
+        $ih = $this->getHeight();
+
+        $ww = $watermark->getWidth();
+        $wh = $watermark->getHeight();
+
+        $y = $settings['align'][0];
+        $x = $settings['align'][1];
+
+        $x = Utils\Dimmensions::getPositionFromReferenceValue($x, $iw, $ww, $settings['padding']);
+        $y = Utils\Dimmensions::getPositionFromReferenceValue($y, $ih, $wh, $settings['padding']);
+
+        $this->image->watermark($watermark->getImage(), $x, $y);
```

Comment:
Yes, I know, and will be return an error if you mix libraries, but if you set the default configuration to libraries and allow an object as parameter, you will can transform the watermark before to be added to the main image.